pkg/compactor/compactor_metrics.go

```diff
@@ -159,6 +161,14 @@ func newCompactorMetricsWithLabels(reg prometheus.Registerer, commonLabels []str
 		Name: "cortex_compactor_remaining_planned_compactions",
 		Help: "Total number of plans that remain to be compacted. Only available with shuffle-sharding strategy",
 	}, commonLabels)
+	m.compactionRetryErrors = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
```

Comment:
We dont have a metric for the PermissionDenied error. Shouldn't we create only one metric and put a reason as label? Then we can expand for the PermissionDenied or other issues in the future.
wdyt?

Reply:
Good suggestion. I have merged those two metrics into one and add `type` label for the metric to indicate if it is retriable, halt, or unauthorized error